### ISSUE TITLE
Training loss remains nan

### ISSUE THREAD

**Issue by jiao0805** [shekkizh/FCN.tensorflow#14]
Hi,

I want to use the code to train my own image data.
And the data image is 512*512 gray image. 
But when I train with them,
the loss remains nan:

Setting up Saver...
Step: 0, Train_loss:nan
2017-03-17 13:55:04.919050 ---> Validation_loss: nan
Step: 10, Train_loss:nan
Step: 20, Train_loss:nan
Step: 30, Train_loss:nan
Step: 40, Train_loss:nan

In the py file BatchDatsetReader I changed the way to read images like:

    def _read_images(self):
        self.__channels = False
        self.images = np.array(
            [np.expand_dims(self._transform(filename['image']), axis=3) for filename in self.files])
But it did not work 

How can I solve this problem to train my gray image ?
And what is the mean of NUM_OF_CLASSESS?

**Comment by shekkizh** [comment]
Each value in the annotation image corresponds to an object in the image. NUM_OF_CLASSESS corresponds to the maximum number of objects to be annotated i.e the max. value in annotation image provided the values are continuous.

Nan can result due to several reasons. Assuming your inputs are all fine, try reducing learning rate.

**Comment by jiao0805** [comment]
Thank you very much
will give a try

**Comment by jiao0805** [comment]
I tried to reduce my learning rate but the loss still remained nan.
I turned on the debug mode and i got the following error.


InvalidArgumentError (see above for traceback): Nan in summary histogram for: inference/conv2_1_w_1
	 [[Node: inference/conv2_1_w_1 = HistogramSummary[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/cpu:0"](inference/conv2_1_w_1/tag, inference/conv2_1_w/read/_107)]]

CONV2_1_W_1 is loaded from the vgg19.mat, isn't it？
Then what makes the node value nan？

**Comment by shekkizh** [comment]
Although the weights are loaded from saved file while training you are changing weights which at some point becomes *nan*. Try learning rates in the order of 10-7 or even smaller to verify if the problem is actually due to large loss terms.

**Comment by shekkizh** [comment]
@jiao0805 Maybe you are facing the same problem discussed in Issue #17

**Comment by jiao0805** [comment]
Yes I solved this problem by change the num of classes。
Thanks a lot and I will close this problem

**Comment by myusernamesxy** [comment]
Hi, I face the same problem with you, but I haven't solved it, may I discuss with you?
